exchanges/exchanges_test.go

```diff
@@ -283,7 +298,11 @@ func (p *fakePoloniexWebsocket) Write(interface{}) error {
 }
 
 func (p *fakePoloniexWebsocket) Close() {
-	close(poloniexDoneChannel)
+	select {
+	case <-poloniexDoneChannel:
+	default:
+		close(poloniexDoneChannel)
```

Comment:
Is Close() possibly called in different places?  I'm trying to understand why this select is needed.  But it is a race since the channel could be closed by another goroutine if Close() is called elsewhere (they could both end up at the `close` statement and one would panic)

Reply:
I think it's unlikely during the test, but easy to add some sequencing.